src/kernel-install/60-ukify.install.in

```diff
@@ -0,0 +1,192 @@
+    except IsADirectoryError:
+        if dir:
+            return
+        raise
+
+def parse_args(args=None):
+    p = argparse.ArgumentParser(
+        description='kernel-install plugin to build a Unified Kernel Image',
+        allow_abbrev=False,
+        usage='''\
+usage: 60-ukify.install COMMAND KERNEL_VERSION ENTRY_DIR KERNEL_IMAGE INITRD…
+       60-ukify.install -h | --help
+''')
+
+    # Suppress printing of usage synopsis on errors
+    p.error = lambda message: p.exit(2, f'{p.prog}: error: {message}\n')
+
+    p.add_argument('command',
+                   metavar='COMMAND',
+                   help="The action to perform. Only 'add' is supported.")
+    p.add_argument('kernel_version',
+                   metavar='KERNEL_VERSION',
+                   help='Kernel version string')
+    p.add_argument('entry_dir',
+                   metavar='ENTRY_DIR',
+                   type=Path,
+                   nargs='?',
+                   help='Type#1 entry directory (ignored)')
+    p.add_argument('kernel_image',
+                   metavar='KERNEL_IMAGE',
+                   type=Path,
+                   nargs='?',
+                   help='Kernel binary')
+    p.add_argument('initrd',
+                   metavar='INITRD…',
+                   type=Path,
+                   nargs='*',
+                   help='Initrd files')
+    p.add_argument('--version',
+                   action='version',
+                   version=f'systemd {__version__}')
+
+    opts = p.parse_args(args)
+
+    if opts.command == 'add':
+        opts.staging_area = Path(os.environ['KERNEL_INSTALL_STAGING_AREA'])
+        path_is_readable(opts.staging_area, dir=True)
+
+        assert opts.kernel_image is not None
+        path_is_readable(opts.kernel_image)
+
+    return opts
+
+def we_are_wanted() -> bool:
+    KERNEL_INSTALL_LAYOUT = os.getenv('KERNEL_INSTALL_LAYOUT')
+
+    if KERNEL_INSTALL_LAYOUT != 'uki':
+        log(f'{KERNEL_INSTALL_LAYOUT=}, quitting.')
+        return False
+
+    KERNEL_INSTALL_UKI_GENERATOR = os.getenv('KERNEL_INSTALL_UKI_GENERATOR')
+
+    if KERNEL_INSTALL_UKI_GENERATOR != 'ukify':
+        log(f'{KERNEL_INSTALL_UKI_GENERATOR=}, quitting.')
+        return False
+
+    log('KERNEL_INSTALL_LAYOUT and KERNEL_INSTALL_UKI_GENERATOR are good')
+    return True
+
+def kernel_cmdline() -> str:
+    entry_token = os.environ['KERNEL_INSTALL_ENTRY_TOKEN']
+    machine_id = os.environ['KERNEL_INSTALL_MACHINE_ID']
+
+    filter = False
+    if root := os.getenv('KERNEL_INSTALL_CONF_ROOT'):
+        cmdline = Path(root) / 'cmdline'
+        if not cmdline.exists():
+            return None
+    else:
+        cmdline = Path('/etc/kernel/cmdline')
+        if not cmdline.exists():
+            cmdline = Path('/usr/lib/kernel/cmdline')
+            if not cmdline.exists():
+                cmdline = Path('/proc/cmdline')
+                filter = True
+                if not cmdline.exists():
+                    return None
```

Comment:
not a fan of such ttocttou iterations. why not:

```
text = None
for f in ('/etc/kernel/cmdline', '/usr/lib/kernel/cmdline', '/proc/cmdline'):
    try:
        text = options.read_text()
    except FileNotFoundError:
        continue

options = text.split()
```

This isn't prone to ttocttou, and shorter too, and less indentation as well!